src/components/Nav.jsx

```diff
@@ -1,39 +1,37 @@
-import React from "react";
-import { Link, useNavigate } from "react-router-dom";
+import React from 'react';
+import { Link, useNavigate } from 'react-router-dom';
 
-export function Nav({ setFilterTopic }) {
+export function Nav() {
+  const navigate = useNavigate();
 
-    const navigate= useNavigate();
+  const handleTopicChange = (topic) => {
+    navigate(`/?topic=${topic}`);
+  };
 
-    const handleTopicChange = (topic) => {
-      setFilterTopic(topic);
-      navigate(`/?topic=${topic}`);
-    };
-  
-    return (
-      <nav>
-        <ul>
-          <li>
-            <Link to="/" onClick={() => handleTopicChange('')}>
-              All Articles
-            </Link>
-          </li>
-          <li>
-            <Link to="/?topic=coding" onClick={() => handleTopicChange('coding')}>
-              Coding Articles
-            </Link>
-          </li>
-          <li>
-            <Link to="/?topic=cooking" onClick={() => handleTopicChange('cooking')}>
-              Cooking Articles
-            </Link>
-          </li>
-          <li>
-            <Link to="/?topic=football" onClick={() => handleTopicChange('football')}>
-              Football Articles
-            </Link>
-          </li>
-        </ul>
-      </nav>
-    );
-  }
\ No newline at end of file
+  return (
+    <nav>
+      <ul>
+        <li>
+          <Link to="/" onClick={() => handleTopicChange('')}>
+            All Articles
+          </Link>
+        </li>
+        <li>
+          <Link to="/?topic=coding" onClick={() => handleTopicChange('coding')}>
+            Coding Articles
+          </Link>
+        </li>
+        <li>
+          <Link to="/?topic=cooking" onClick={() => handleTopicChange('cooking')}>
+            Cooking Articles
+          </Link>
+        </li>
+        <li>
+          <Link to="/?topic=football" onClick={() => handleTopicChange('football')}>
+            Football Articles
+          </Link>
+        </li>
+      </ul>
+    </nav>
```

Comment:
This works for the database in this exact state, but it could be more flexible. The topic links are currently hardcoded which isn't ideal - what if a topic were to change? For things like this, we always should prefer a dynamic approach.

_Remember that you can also GET topics from the api_